platform/CThunk.h

```diff
@@ -37,7 +37,8 @@
 #define CTHUNK_ADDRESS 1
 #define CTHUNK_VARIABLES volatile uint32_t code[2]
 
-#if (defined(__CORTEX_M3) || defined(__CORTEX_M4) || defined(__CORTEX_M7) || defined(__CORTEX_A9))
+#if (defined(__CORTEX_M3) || defined(__CORTEX_M4) || defined(__CORTEX_M7) || defined(__CORTEX_A9) \
+    || defined(__CORTEX_M23) || defined(__CORTEX_M33))
```

Comment:
Armv8-M architecture reference manual has new term now for Thumb instructions. It says v8-M variant supports T32 instruction set and does not distinguish between mainline (M-33) and baseline (M-23) architecture.
All instructions are  decoded as T32 (not as 16-bit Thumb has to be aligned to half word).

Also, callbacks in test were successful.